app/packages/app/src/useEvents/utils.ts

```diff
@@ -24,4 +24,32 @@ export const processState = (
     toCamelCase(state.selected_labels) as State.SelectedLabel[]
   );
   state.spaces && setter("sessionSpaces", state.spaces);
+
+  const visibility = state.field_visibility_stage;
+  const visibilityStage = visibility
+    ? {
+        _cls: visibility.cls,
+        kwargs: [
+          ["field_names", visibility.field_names],
+          ["_allow_missing", true],
+        ],
+      }
+    : null;
+  const stageAtom = visibility
+    ? {
+        cls: visibility.cls,
+        kwargs: {
+          field_names: visibility.field_names,
+          allow_missing: true,
+        },
+      }
+    : undefined;
+
+  stageAtom && setter("fieldVisibilityStage", stageAtom);
+  return env().VITE_NO_STATE
+    ? { view: [], extendedStages: [] }
+    : {
+        view: state.view || [],
+        extendedStages: visibilityStage ? [visibilityStage as State.Stage] : [],
+      };
```

Comment:
For testing purposes, when `env().VITE_NO_STATE` is true, we shouldn't set the field visibility stage
```suggestion
  if (env().VITE_NO_STATE) {
    return { view: [], extendedStages: [] }
  }
  stageAtom && setter("fieldVisibilityStage", stageAtom);
  return {
      view: state.view || [],
      extendedStages: visibilityStage ? [visibilityStage as State.Stage] : [],
   };
```